src/Bicep.Core/Intermediate/ExpressionRewriteVisitor.cs

```diff
@@ -789,6 +790,28 @@ private bool TryRewriteStrict<TExpression>(ImmutableArray<TExpression> expressio
         return hasChanges;
     }
 
+    private bool TryRewriteDictionaryStrict<TExpression>(ImmutableDictionary<string, TExpression>? dictionary, [NotNullIfNotNull("dictionary")] out ImmutableDictionary<string, TExpression>? newDictionary)
+        where TExpression : ArrayExpression
+    {
+        if (dictionary is null)
+        {
+            newDictionary = null;
+            return false;
+        }
+
+        var hasChanges = false;
+        var newDictionaryList = ImmutableDictionary.CreateBuilder<string, TExpression>();
```

Comment:
Just to be safe (in case the original dictionary was instantiated with a custom comparer):
```suggestion
        var newDictionaryList = ImmutableDictionary.CreateBuilder<string, TExpression>(dictionary.KeyComparer, dictionary.ValueComparer);
```